src/Ads/AssetSuggestionsService.php

```diff
@@ -0,0 +1,128 @@
+<?php
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\GoogleListingsAndAds\Ads;
+
+use Automattic\WooCommerce\GoogleListingsAndAds\Infrastructure\Service;
+use Automattic\WooCommerce\GoogleListingsAndAds\Proxies\WP;
+
+/**
+ * Class AssetSuggestionsService
+ *
+ * Suggest assets and possible final URLs.
+ *
+ * @since x.x.x
+ *
+ * @package Automattic\WooCommerce\GoogleListingsAndAds\Ads
+ */
+class AssetSuggestionsService implements Service {
+
+	/**
+	 * AssetSuggestionsService constructor.
+	 *
+	 * @param WP $wp
+	 */
+	public function __construct( WP $wp ) {
+		$this->wp = $wp;
+	}
+
+	/**
+	 * Get posts that can be used to suggest assets
+	 *
+	 * @return array
+	 */
+	protected function get_post_suggestions(): array {
+		$post_suggestions    = [];
+		$excluded_post_types = [ 'attachment' ];
+
+		$post_types = $this->wp->get_post_types(
+			[
+				'exclude_from_search' => false,
+				'public'              => true,
+			]
+		);
+
+		// Exclude attachment post_type
+		$filtered_post_types = array_diff( $post_types, $excluded_post_types );
+
+		$args = [
+			'post_type'      => $filtered_post_types,
+			'posts_per_page' => -1,
```

Comment:
❓ Don't take this as a 100% assumption... but are we sure we want to return all the posts for a site? (+ all terms) That could be a huge number of posts/terms. I'm unsure about the potential performance leaks on this approach. 

As far as I read in the project, this is going to be a selector in where the user can type the URL or Post/Term name. 
So I wonder if we can just return a limited number of Posts/Terms and maybe have an extra endpoint for querying them. 🤔 

What are your thoughts about this?

Reply:
Yes, I had the same concern, I talked to @eason9487  about this last week, and for now, the idea is to return the complete list of URLs and depending on how the WP/WC component works we could add some type of matching URLs function depending on the input. Also, the final design still needs to be determined. See comment here: pcTzPl-1eS-p2#comment-2048